packages/massa-web3/src/experimental/basicElements/mas.ts

```diff
@@ -122,10 +122,23 @@ export function toString(
   let decimalPart = valueString.slice(-NB_DECIMALS).padStart(NB_DECIMALS, '0')
 
   if (decimalPlaces === null) {
+    // If decimalPlaces is null, remove trailing zeros from the decimal part
     decimalPart = decimalPart.replace(/0+$/, '')
-  } else {
-    // TODO - Round the number to the desired number of decimal places
-    decimalPart = decimalPart.slice(ZERO, decimalPlaces)
+  } else if (decimalPlaces < NB_DECIMALS) {
+    // If decimalPlaces is specified and less than NB_DECIMALS, handle rounding
+    const roundingDigit = parseInt(decimalPart[decimalPlaces])
+
+    if (roundingDigit >= FIVE) {
+      // If the rounding digit is 5 or greater, round up
+      decimalPart = (
+        BigInt(decimalPart.slice(ZERO, decimalPlaces)) + BigInt(ONE)
+      )
+        .toString()
+        .padStart(decimalPlaces, '0')
+    } else {
+      // Otherwise, truncate the decimal part to the specified number of places
+      decimalPart = decimalPart.slice(ZERO, decimalPlaces)
+    }
```

Comment:
This code is getting to complicated, you should use something like [decimal.js](https://github.com/MikeMcl/decimal.js).